content/articles/php-sessions-and-cookies-to-create-a-secure-remember-me-section-in-a-login-script/Index.md

```diff
@@ -0,0 +1,433 @@
+
+        // Set Auth Cookies if 'Remember Me' checked
+        if (! empty($_POST["remember"])) {
+            setcookie("member_login", $username, $cookie_expiration_time);
+
+            $random_password = $util->getToken(16);
+            setcookie("random_password", $random_password, $cookie_expiration_time);
+
+            $random_selector = $util->getToken(32);
+            setcookie("random_selector", $random_selector, $cookie_expiration_time);
+
+            $random_password_hash = password_hash($random_password, PASSWORD_DEFAULT);
+            $random_selector_hash = password_hash($random_selector, PASSWORD_DEFAULT);
+
+            $expiry_date = date("Y-m-d H:i:s", $cookie_expiration_time);
+
+            // mark existing token as expired
+            $userToken = $auth->getTokenByUsername($username, 0);
+            if (! empty($userToken[0]["id"])) {
+                $auth->markAsExpired($userToken[0]["id"]);
+            }
+            // Insert new token
+            $auth->insertToken($username, $random_password_hash, $random_selector_hash, $expiry_date);
+        } else {
+            $util->clearAuthCookie();
+        }
+        $util->redirect("dashboard.php");
+    } else {
+        $message = "Invalid Login";
+    }
+}
+?>
+```
+
+- HTML code to display login form with a "Remember Me" option.
+
+```html
+<form action="" method="post" id="frmLogin">
+    <div class="error-message"><?php if(isset($message)) { echo $message; } ?></div>
+    <div class="field-group">
+        <div>
+            <label for="login">Username</label>
+        </div>
+        <div>
+            <input name="member_name" type="text"
+                value="<?php if(isset($_COOKIE["member_login"])) { echo $_COOKIE["member_login"]; } ?>"
+                class="input-field">
+        </div>
+    </div>
+    <div class="field-group">
+        <div>
+            <label for="password">Password</label>
+        </div>
+        <div>
+            <input name="member_password" type="password"
+                value="<?php if(isset($_COOKIE["member_password"])) { echo $_COOKIE["member_password"]; } ?>"
+                class="input-field">
+        </div>
+    </div>
+    <div class="field-group">
+        <div>
+            <input type="checkbox" name="remember" id="remember"
+                <?php if(isset($_COOKIE["member_login"])) { ?> checked
+                <?php } ?> /> <label for="remember-me">Remember me</label>
+        </div>
+    </div>
+    <div class="field-group">
+        <div>
+            <input type="submit" name="login" value="Login"
+                class="form-submit-button"></span>
+        </div>
+    </div>
+</form>
+```
+
+### Validate Remembered Login with PHP Session and Cookies
+
+---
+
+authCookieSessionValidate.php, a PHP page, contains code for validating the logged-in state based on session and cookie data. For applications that require user authentication, it appears at the beginning of the application page.

```

Comment:
Find a way to put emphasis on file names throughout your article. And again, do this where applicable.